engine/src/main/java/org/pentaho/di/www/GetStatusServlet.java

```diff
@@ -253,74 +251,78 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
           DocumentBuilder db = dbFactory.newDocumentBuilder();
           Document doc = db.parse( f );
           String themeName = doc.getElementsByTagName( "default-theme" ).item( 0 ).getTextContent();
 
           // Get theme CSS file
-          String themePath = "..\\..\\pentaho-solutions\\system\\common-ui\\resources\\themes\\" + themeName + "\\";
-          File themeDir = new File( themePath );
+          String themeCss = ".." + File.separator + ".." + File.separator
+              + "pentaho-solutions" + File.separator + "system" + File.separator
+              + "common-ui" + File.separator + "resources" + File.separator
+              + "themes" + File.separator + themeName + File.separator;
+          File themeDir = new File( themeCss );
           for ( File fName : themeDir.listFiles() ) {
             if ( fName.getName().contains( ".css" ) ) {
-              themePath += fName.getName();
+              themeCss = fName.getName();
               break;
             }
           }
 
           // Get mantle theme CSS file
-          String mantlePath = "..\\webapps\\pentaho\\mantle\\themes\\" + themeName + "\\";
-          File mantleDir = new File( mantlePath );
-          for ( File fName : mantleDir.listFiles() ) {
+          String mantleThemeCss = ".." + File.separator + "webapps" + File.separator
+              + "pentaho" + File.separator + "mantle" + File.separator
+              + "themes" + File.separator + themeName + File.separator;
+          File mantleThemeDir = new File( mantleThemeCss );
+          for ( File fName : mantleThemeDir.listFiles() ) {
             if ( fName.getName().contains( ".css" ) ) {
-              mantlePath += fName.getName();
+              mantleThemeCss = fName.getName();
               break;
             }
           }
 
-          BufferedReader styleBr = new BufferedReader( new FileReader( themePath ) ); // themePath with appended CSS filename
-          StringBuilder sb = new StringBuilder();
-
-          // Start to read in CSS and write HTML
-          out.println( "<STYLE>" );
-
-          String line = styleBr.readLine();
-          while ( line != null ) {
-            sb.append( line );
-            line = styleBr.readLine();
-          }
-
-          out.println( sb.toString() );
-          out.println( "</STYLE>" );
-          out.println( "<STYLE>" );
-
-          BufferedReader mantleBr = new BufferedReader( new FileReader( mantlePath ) );
-          sb.delete( 0, sb.length() ); // clear builder
-          line = mantleBr.readLine();
-          while ( line != null ) {
-            sb.append( line );
-            line = mantleBr.readLine();
-          }
-
-          styleBr.close();
-          out.println( sb.toString() );
-          out.println( "</STYLE>" );
+          out.println( "<link rel=\"stylesheet\" type=\"text/css\" href=\"/pentaho/content/common-ui/resources/themes/" + themeName + "/" + themeCss + "\"/>" );
```

Comment:
How will the styles be handle in the context of standalone carte when pentaho server is not available?

Reply:
I guess this code doesn't account for that, but should a standalone carte even display themes? If so I would think we'd need to copy all of the Pentaho themes located in the pentaho-solutions/system/common-ui folder over to PDI and then add more logic into this to look into both locations.